base/ca/src/com/netscape/ca/CAService.java

```diff
@@ -896,25 +899,70 @@ private X509CertImpl issueX509Cert(
                     logger.debug(method + " about to check CT poison");
                     Extension ctPoison = (Extension) exts.get("1.3.6.1.4.1.11129.2.4.3");
                     if ( ctPoison == null) {
+                        // this just means that CT is enabled, but profile doen'st,
+                        // which we honor
                         logger.debug(method + " ctPoison not found, no CT processing needed");
                     } else {
                         logger.debug(method + " ctPoison found");
+                        // debug print with poison ext
+                        // printExtensions(exts);
+
+                        // remove then put back to assure relocating poison ext
+                        // to end of the map
+                        exts.delete("1.3.6.1.4.1.11129.2.4.3");
```

Comment:
why do we want to remove and add back? Why not declare a new `X509CertInfo` and copy it?

````
X509CertInfo certWithPoisonRemoved = certi;
exts.delete("1.3.6.1.4.1.11129.2.4.3");
...
certi.set(X509CertInfo.EXTENSIONS, exts);
````

Reply:
yes. That's one of the things I tried earlier.  The latest manipulation with exts is just so that I could printExtensions and observer the "order" (see my next reply to your comment).  These are all going to be cleared off later when I switch over to LinkedHashMap officially.